xt_imager.py

```diff
@@ -134,8 +143,14 @@ def do_flash_image(args, tftp_root):
     # - read X MB chunk from image file
     # - save chunk to file in tftp root
     # - tell u-boot to 'tftp-and-emmc' chunk
-    while bytes_sent < image_size:
-        data = f_img.read(chunk_size_in_bytes)
+    while True:
+        if use_lzma:
+            data = f_lzma.read(chunk_size_in_bytes)
```

Comment:
So what is the difference? Python has a concept of "file-like object".  `lzma.open()` returns this type of object.

You can use `f_img` in both cases. Just initialize it with either `open` or `lzma.open`.

Reply:
Thanks, reworked.